creator/ingest_runs/common/model.py

```diff
@@ -0,0 +1,182 @@
+import hashlib
+import uuid
+import re
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+from django_fsm import FSMField, transition
+
+from creator.jobs.models import JobLog
+from creator.utils import stop_job
+
+User = get_user_model()
+
+
+def camel_to_snake(camel_str):
+    """
+    Convert CamelCase to snake_case
+    """
+    return re.sub(r'(?<!^)(?=[A-Z])', '_', camel_str).lower()
+
+
+def hash_versions(versions):
+    """
+    Uniquely identify a set of file Versions by computing an MD5 hash digest
+    using the primary keys of the Versions
+    """
+    version_id_str = "".join(sorted(v.kf_id for v in versions))
+    return hashlib.md5(version_id_str.encode("utf-8")).hexdigest()
+
+
+class State(object):
```

Comment:
Would this be better served by an Enum? We also tend to omit the `object` as it's already implied.

Reply:
Ehh, normally yes, but I use the State class a lot to refer to the states and I just hate having to do `State.RUNNING.value` vs `State.RUNNING` everywhere.